py/picca/fitter2/xi.py

```diff
@@ -119,9 +121,56 @@ def xi_qso_radiation(r, mu, k, pk_lin, pk_func, tracer1, tracer2, ell_max=None,
     return xi_drp(r, mu, k, pk_lin, pk_func, tracer1, tracer2, ell_max, **pars) + xi_rad
 
 ### Growth factor evolution
-def growth_function_no_de(z, zref = None, **kwargs):
+def growth_factor_no_de(z, zref=None, **kwargs):
     return (1+zref)/(1.+z)
 
+def cache_growth_factor_de(function):
+    cache = {}
+    def wrapper(*args, **kwargs):
+        Om = kwargs['Om']
+        OL = kwargs['OL']
+        pair = ('Om', 'OL')
+        if pair not in cache.keys() or not sp.allclose(cache[pair], (Om,OL)):
+            cache[pair] = (Om, OL)
+            cache[1] = cached_growth_factor_de(*args, **kwargs)
+        
+        return cache[1](args[0])/cache[1](kwargs['zref'])
+            
+    return wrapper
+
+@cache_growth_factor_de
+def growth_factor_de(*args, **kwargs):
+    return cached_growth_factor_de(*args, **kwargs)
+
+def cached_growth_factor_de(z, zref = None, **kwargs):
+    '''
+    Implements eq. 7.77 from S. Dodelson's Modern Cosmology book
+    '''
+    print('Calculating growth factor for Om = {}, OL = {}'.format(kwargs['Om'], kwargs['OL']))
+
+    def hubble(z, Om=0.3, OL=0.7):
```

Comment:
maybe it would be better to have the values `Om=0.315` and `OL=1.-Om=0.685` by default, to be consistent with the rest of picca.

Reply:
good point. Actually I will remove the default values to force the user to specify them

Reply:
In fact, we should use the same `Om`, `OL` as what was used to calculate the `P(k)`, shouldn't we? But I don't think this is written in the input p(k) file. I think we should write in the header of the P(k) fits file all the parameters from the .ini that was used to run camb...

Reply:
I can take care of adding those values to the P(k) fits file.